### DISCUSSION TITLE
Welcome to hummingbird Discussions!

### DISCUSSION THREAD

**Discussion by adam-fowler** [hummingbird-project/hummingbird#48] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
  * Please read out [Code of conduct](https://github.com/hummingbird-project/hummingbird/blob/main/CODE_OF_CONDUCT.md) before posting here. 

Hi,

My name is Adam Fowler and I am the main developer of Hummingbird. I started this project because I wanted to provide an alternative to our big brother Vapor. The project is in its early days at the moment and I would be very interested in hearing feedback. If you would like to contribute to the project, either through issues, conversations here or pull requests that would be even better.

Cheers 
Adam

**Comment by nonfungibletunji** [comment]
Looks clean, just starting looking at the docs. I have been trying to been trying to read from external graphql API service, using a swift server deployed app. working on an example GraphQL server example hummingbird-graphql-kit ????

**Reply by adam-fowler** [reply]
Unfortunately no hummingbird GraphQL framework available yet. You could look at https://github.com/GraphQLSwift/Graphiti though.

**Comment by nonfungibletunji** [comment]
That’s the plan

S Tunji Turner

**Comment by nonfungibletunji** [comment]
I read some more of the documentation, what are the differences between this and the Vapor : Server Side Software?

**Reply by adam-fowler** [reply]
Hi, yes Vapor and Hummingbird do have a number of similarities. A router, middleware are pretty standard across many application frameworks, but the add additional features like extending `HBApplication`, `HBRequest` is very similar to Vapor, although done through a slightly different manner, there are similarities elsewhere because I have come from a position of using Vapor and can't avoid being influenced by what has come before.

The main difference between the two is Vapor is a big beast it provides almost everything in one library regardless of whether you need it. The idea behind Hummingbird is to provide only what the user needs. If they don't want TLS don't force them to compile TLS libraries, web sockets, HTTP2 etc. 

I split the main code of hummingbird into two repositories so that if someone doesn't like the application framework and only wants the HTTP server they can take the server elements and write their own framework on top.

**Comment by nonfungibletunji** [comment]
Thanks for the detailed response

S Tunji Turner

**Comment by cyrildewit** [comment]
Hi @adam-fowler, @Joannis and all the other awesome people that are missing here,

I am not a Swift developer, but just wanted to take a moment to express my gratitude for the effort that has gone into the projects! I have primarily explored the documentation, read articles on the internet and skimmed through the source code of the initiatives. I’m not fully qualified to give a detailed opinion, but everything I’ve seen so far makes it look like the initiative is clean, well-designed, and very flexible.

I am excited for the project (and swift) and I am looking forward to future updates. Kudos 👍
